pkg/virt-handler/vm.go

```diff
@@ -66,6 +69,60 @@ type VMHandlerDispatch struct {
 	host          string
 }
 
+func (d *VMHandlerDispatch) getVMNodeAddress(vm *v1.VM) (string, error) {
+	node, err := d.clientset.CoreV1().Nodes().Get(vm.Status.NodeName, metav1.GetOptions{})
+	if err != nil {
+		logging.DefaultLogger().Error().Reason(err).Msgf("fetching source node %s failed", vm.Status.NodeName)
+		return "", err
+	}
+
+	addrStr := ""
+	for _, addr := range node.Status.Addresses {
+		if (addr.Type == kubev1.NodeInternalIP) && (addrStr == "") {
+			addrStr = addr.Address
+			break
+		}
+	}
+	if addrStr == "" {
+		err := fmt.Errorf("VM node is unreachable")
+		logging.DefaultLogger().Error().Msg("VM node is unreachable")
+		return "", err
+	}
+
+	return addrStr, nil
+}
+
+func (d *VMHandlerDispatch) updateVMStatus(vm *v1.VM, cfg *api.DomainSpec) error {
+	obj, err := kubeapi.Scheme.Copy(vm)
+	if err != nil {
+		return err
+	}
+	vm = obj.(*v1.VM)
+	vm.Status.Phase = v1.Running
+
+	vm.Status.Graphics = []v1.VMGraphics{}
+
+	podIP, err := d.getVMNodeAddress(vm)
```

Comment:
Right - so we probably need to do this once that move is done. but this is outside of scope of this PR iiuic.